### PR TITLE
Team6

### PR DESCRIPTION
**Pull request by rahulsagar500**
1. Added Background for MainMenu
2. Added DragAndDrop
3. Added PopUp for DragAndDrop
4. Added GameOver PopUp after health becomes 0.

### CODE REVIEW

**Comment by rahulsagar500** [comment]
1. Added Background for MainMenu
2. Added DragAndDrop
3. Added PopUp for DragAndDrop
4. Added GameOver PopUp after health becomes 0.

**Comment by rahulsagar500** [comment]
1. Added Background for MainMenu
2. Added DragAndDrop
3. Added PopUp for DragAndDrop
4. Added GameOver PopUp after health becomes 0.

File: source/core/src/main/com/csse3200/game/components/gamearea/GameAreaDisplay.java [hunk 1]
```diff
@@ -1,44 +1,316 @@
...
 
 /**
  * Displays the name of the current game area.
  */
 public class GameAreaDisplay extends UIComponent {
+  private static final Logger logger = LoggerFactory.getLogger(GameAreaDisplay.class);
+  private static final float Z_INDEX = 2f;
+  // Dialog for displaying tower details
+
+  private Vector2[] towerPositions1;
```
Comment (The-AhmadAA) [inline source/core/src/main/com/csse3200/game/components/gamearea/GameAreaDisplay.java:36]: It's not clear what is happening in this method. What are all the tower variables here for?

File: source/core/src/main/com/csse3200/game/components/gamearea/GameAreaDisplay.java [hunk 2]
```diff
@@ -1,44 +1,316 @@
...
+
+        public void touchUp(InputEvent event, float x, float y, int pointer, int button) {
+          towerDetailsDialog1.hide();
+        }
+      });
+    }
+    for (int i = 0; i < 2; i++) {
+      // Use "building1" for the first tower and "building2" for the second tower
+      skin.add("default", new Label.LabelStyle(new BitmapFont(), Color.WHITE));
+      skin.add("building2", new Texture("images/towers/turret_deployed.png"));
```
Comment (The-AhmadAA) [inline source/core/src/main/com/csse3200/game/components/gamearea/GameAreaDisplay.java:146]: this "turret_deployed.png" image is no longer being used for anything in game

File: source/core/src/main/com/csse3200/game/components/player/PlayerStatsDisplay.java
Comment (The-AhmadAA) [inline source/core/src/main/com/csse3200/game/components/player/PlayerStatsDisplay.java:1]: The game over functionality is already covered by ticket #132 and #80

**Review by The-AhmadAA — CHANGES_REQUESTED** [review]
The game over functionality in particular is redundant and covered by other task tickets. Please disable this so it does not interfere with game over funcitionality

File: source/core/src/main/com/csse3200/game/components/player/PlayerStatsDisplay.java
Comment (rahulsagar500) [reply source/core/src/main/com/csse3200/game/components/player/PlayerStatsDisplay.java:1]: hey @The-AhmadAA this feature was our task for team 6 and we did discuss this in the studio standups as well. The only problem was that we did not know the condition of the game over yet and forthe time being assigned the screen visibility when the health of player becomes 0.

File: source/core/src/main/com/csse3200/game/components/gamearea/GameAreaDisplay.java (hunk 2 above)
Comment (rahulsagar500) [reply source/core/src/main/com/csse3200/game/components/gamearea/GameAreaDisplay.java:146]: sorry for that, We worked in our remote rep, changing that now

File: source/core/src/main/com/csse3200/game/components/gamearea/GameAreaDisplay.java (hunk 1 above)
Comment (rahulsagar500) [reply source/core/src/main/com/csse3200/game/components/gamearea/GameAreaDisplay.java:36]: this is just a structural design for our functionality of drag and drop, we will integrate it with the real entities for the upcoming sprint

**Pull request merged**